### PR TITLE
test(daemon): Add `makeBundle` test

### PR DESCRIPTION
**Pull request by rekmarks**
Closes: #2024 

Tests confined bundles using `makeBundle`. Adds a utility to make using `makeBundle` significantly easier in tests for the daemon.

### CODE REVIEW

File: packages/daemon/test/test-endo.js [hunk 1]
```diff
@@ -41,6 +42,36 @@ const makeLocator = (...root) => {
   };
 };
 
+// The id of the next bundle to make.
+let bundleId = 0;
+const textEncoder = new TextEncoder();
+
+// TODO: We should be able to use {import('../src/types').EndoHost} for `host`,
+// but when laundered through `E()` it becomes `never`.
```
Comment (rekmarks) [inline packages/daemon/test/test-endo.js:50]: @kriskowal @kumavis is this something you've encountered before? Is it user error?

Comment (kriskowal) [reply packages/daemon/test/test-endo.js:50]: No. The `.js` extension may be required (even though it is a `.d.ts`)

**Review by kriskowal — APPROVED** [review]
Very interested in obviating the need for temporary names. Maybe we can eventually use dotted paths into mailbox messages and log commands.

File: packages/daemon/test/test-endo.js (hunk 1 above)
Comment (rekmarks) [reply packages/daemon/test/test-endo.js:50]: This does not appear to make a difference on my system. The error is visible in my editor (VSCode) using the repo version of TypeScript (`5.3.3`). ESLint does not appear to be configured to validate JSDoc types (although we could make it so).

**Pull request merged**